pkg/apis/core/validation/validation_test.go

```diff
@@ -7753,88 +7748,110 @@ func TestValidateResizePolicy(t *testing.T) {
+			Errors:           field.ErrorList{field.Invalid(field.NewPath("field"), core.ResourceResizeRestartPolicy("RestartContainer"), `resize restart policy "RestartContainer" cannot be used with pod restart policy "Never"`)},
 			PodRestartPolicy: "Never",
 		},
 		"InvalidMemoryCPUPolicyWithPodRestartPolicy": {
 			PolicyList: []core.ContainerResizePolicy{
 				{ResourceName: "cpu", RestartPolicy: "RestartContainer"},
 				{ResourceName: "memory", RestartPolicy: "RestartContainer"},
 			},
-			ExpectError:      true,
-			Errors:           field.ErrorList{field.Invalid(field.NewPath("field"), core.ResourceResizeRestartPolicy("RestartContainer"), "must be 'NotRequired' when `restartPolicy` is 'Never'"), field.Invalid(field.NewPath("field"), core.ResourceResizeRestartPolicy("RestartContainer"), "must be 'NotRequired' when `restartPolicy` is 'Never'")},
+			Errors:           field.ErrorList{field.Invalid(field.NewPath("field"), core.ResourceResizeRestartPolicy("RestartContainer"), `resize restart policy "RestartContainer" cannot be used with pod restart policy "Never"`), field.Invalid(field.NewPath("field"), core.ResourceResizeRestartPolicy("RestartContainer"), `resize restart policy "RestartContainer" cannot be used with pod restart policy "Never"`)},
 			PodRestartPolicy: "Never",
 		},
 		"ValidMemoryCPUPolicyWithPodRestartPolicy": {
 			PolicyList: []core.ContainerResizePolicy{
-				{ResourceName: "cpu", RestartPolicy: "NotRequired"},
-				{ResourceName: "memory", RestartPolicy: "NotRequired"},
+				{ResourceName: "cpu", RestartPolicy: "PreferNoRestart"},
+				{ResourceName: "memory", RestartPolicy: "PreferNoRestart"},
 			},
-			ExpectError:      false,
 			Errors:           nil,
 			PodRestartPolicy: "Never",
 		},
 	}
+	// Mixin deprecated test cases.
+	const notRequiredCaseSuffix = "(NotRequired)"
 	for k, v := range testCases {
-		errs := validateResizePolicy(v.PolicyList, field.NewPath("field"), &v.PodRestartPolicy)
-		if !v.ExpectError && len(errs) > 0 {
-			t.Errorf("Testcase %s - expected success, got error: %+v", k, errs)
+		if strings.HasSuffix(k, notRequiredCaseSuffix) {
+			continue // Since we're modifying the map while iterating, we may see a case we added. Just skip it.
 		}
-		if v.ExpectError {
-			if len(errs) == 0 {
-				t.Errorf("Testcase %s - expected error, got success", k)
-			}
-			delta := cmp.Diff(errs, v.Errors)
-			if delta != "" {
-				t.Errorf("Testcase %s - expected errors '%v', got '%v', diff: '%v'", k, v.Errors, errs, delta)
+		notRequiredPolicies := []core.ContainerResizePolicy{}
+		hadPreferNoRestart := false
+		for _, p := range v.PolicyList {
+			if p.RestartPolicy == core.ResizeRestartPolicyPreferNoRestart {
+				p.RestartPolicy = core.ResizeRestartPolicyNotRequired
+				hadPreferNoRestart = true
 			}
+			notRequiredPolicies = append(notRequiredPolicies, p)
```

Comment:
This adds all entries, even "RestartContainer" to a list called "notRequiredPolicies" - that seems fishy and there's no comment explaining why...

This code has reached the level of complexity where I would want a test for it.  

I know it feels redundant to duplicate table entries, but it's WAY easier to convince me that it is correct.  Or alternately, in the "for each testcase" loop, scan `v.PolicyList` and set all NotRequired to PreferNoRestart, run the test, then set all PreferNoRestart to NotRequired and trun it again - should get the same results, right? 

Again, way easier to comprehend